vnext/Microsoft.ReactNative.Cxx/PropertyBag.h

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#ifndef MICROSOFT_REACTNATIVE_PROPERTYBAG
+#define MICROSOFT_REACTNATIVE_PROPERTYBAG
+
+//!
+//! PropertyBag is a thread-safe storage of property values.
+//! Internally the value is IInspectable and the key is a name object that represents an
+//! atomized string name. Each name object is defined in the context of a namespace object.
+//! The null or empty namespace object is a global namespace object.
+//! The property name is unique for the same namespace object.
+//! Different namespaces may have properties with the same local names.
+//!
+//! The PropertyBag struct is a wrapper around the ABI-safe IReactPropertyBag interface.
+//! The IReactPropertyBag represents all values as IInspectable object which can wrap any type.
+//! On top of the untyped IReactPropertyBag values, the PropertyBag offers a set of typed property accessors:
+//! Get, GetOrCreate, Remove, and Set.
+//!
+//! To simplify access to properties we offer PropertyName type that helps to
+//! wrap up property name and type into one variable.
+//!
+//! For example, we can define a property to access an integer value:
+//!
+//!     const React::PropertyName<int> MyIntProperty{"MyInt"};
+//!
+//! then we can use it to set property in application settings during initialization:
+//!
+//!     settings.ApplicationProperties().Set(MyIntProperty, 42);
```

Comment:
>ApplicationProperties [](start = 17, length = 21)

Nit: Is it really called "ApplicationProperties"?

Reply:
Good catch! I changed the name to GlobalProperties after I wrote this comment.

Reply:
Renamed to `Properties`.